src/app/intermediate.rs

```diff
@@ -0,0 +1,114 @@
+use std::sync::Arc;
+
+use nysa::global as bus;
+
+use crate::app::{lobby, paint, AppState};
+use crate::assets::Assets;
+use crate::cli;
+use crate::common::{Error, Fatal};
+use crate::config::config;
+use crate::net::{
+   peer::{self, Peer},
+   socket::SocketSystem,
+};
+
+pub struct State {
+   assets: Box<Assets>,
+   socket_system: Arc<SocketSystem>,
+   peer: Option<Peer>,
+}
+
+impl State {
+   pub fn new(
+      cli: cli::Cli,
+      assets: Box<Assets>,
+      socket_system: Arc<SocketSystem>,
+   ) -> Box<dyn AppState> {
+      match cli.command {
+         Some(cli::Commands::HostRoom) => {
+            let peer = Some(Peer::host(
+               Arc::clone(&socket_system),
+               &config().lobby.nickname,
+               &config().lobby.relay,
+            ));
+
+            Box::new(Self {
+               assets,
+               socket_system,
+               peer,
+            })
+         }
+         Some(cli::Commands::JoinRoom { room_id }) => {
+            let peer = Some(Peer::join(
+               Arc::clone(&socket_system),
+               &config().lobby.nickname,
+               &config().lobby.relay,
+               room_id,
+            ));
+
+            Box::new(Self {
+               assets,
+               socket_system,
+               peer,
+            })
+         }
+         _ => Box::new(lobby::State::new(assets, Arc::clone(&socket_system))),
+      }
+   }
+}
+
+impl AppState for State {
+   fn process(&mut self, _: super::StateArgs) {
+      if let Some(peer) = &mut self.peer {
+         catch!(peer.communicate());
+      }
+
+      for message in &bus::retrieve_all::<Error>() {
+         let error = message.consume().0;
+         tracing::error!("error: {:?}", error);
+      }
+      for message in &bus::retrieve_all::<Fatal>() {
+         let fatal = message.consume().0;
+         tracing::error!("fatal: {:?}", fatal);
+      }
+   }
+
+   fn next_state(self: Box<Self>, renderer: &mut crate::backend::Backend) -> Box<dyn AppState> {
```

Comment:
Please `use` `crate::backend::Backend` and `super::StateArgs` such that qualified names are not used for arguments here, as they're unnecessarily verbose.